guacamole-core/src/main/java/io/github/rczyzewski/guacamole/ddb/MappedScanExpression.java

```diff
@@ -0,0 +1,67 @@
+package io.github.rczyzewski.guacamole.ddb;
+
+import io.github.rczyzewski.guacamole.ddb.mapper.ConsecutiveIdGenerator;
+import io.github.rczyzewski.guacamole.ddb.mapper.ExpressionGenerator;
+import io.github.rczyzewski.guacamole.ddb.mapper.LiveMappingDescription;
+import io.github.rczyzewski.guacamole.ddb.mapper.LogicalExpression;
+import lombok.AllArgsConstructor;
+import software.amazon.awssdk.services.dynamodb.model.AttributeValue;
+import software.amazon.awssdk.services.dynamodb.model.ScanRequest;
+import software.amazon.awssdk.services.dynamodb.model.Select;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+
+@AllArgsConstructor
+public class MappedScanExpression<T, G extends ExpressionGenerator<T>>{
+    private final G generator;
+    private final String tableName;
+    private final Map<String, AttributeValue> keys;
+    private final LogicalExpression<T> condition;
+    private final LiveMappingDescription<T> liveMappingDescription;
+    private final ConsecutiveIdGenerator idGenerator = ConsecutiveIdGenerator.builder().base("ABCDE").build();
+
+    public MappedScanExpression<T, G> withCondition(Function<G, LogicalExpression<T>> condition)
+    {
+
+        return this.condition != null ? this : new MappedScanExpression<>(
```

Comment:
the initial part with preparing expression is the same for delete/update/scanning

Reply:
done